src/main/java/me/gaminglounge/guiapi/GuiFromMap.java

```diff
@@ -0,0 +1,92 @@
+package me.gaminglounge.guiapi;
+
+import java.util.Map;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import me.gaminglounge.itembuilder.ItemBuilder;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+
+public class GuiFromMap implements InventoryHolder {
+    private final MiniMessage mm = MiniMessage.miniMessage();
+
+    private Inventory inv;
+    private final InventoryHolder holder;
+    private Component inventoryName;
+    private final int row;
+    private Map<Integer, ItemStack> items;
+    private boolean fill;
+
+    /**
+     * 
+     * @param holder inventoryholder
+     * @param row    how many rows the gui shoud have
+     * 
+     */
+    public GuiFromMap(InventoryHolder holder, int row) {
+        this.row = row;
+        this.holder = holder;
+        this.fill = true;
```

Comment:
Could be handled via the other constructor.
```suggestion
        this(holder, row, true);
```